extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/AbstractMongoClientProducer.java

```diff
@@ -295,6 +297,13 @@ private MongoClientSettings createMongoConfiguration(MongoClientConfig config) {
         if (config.readPreference.isPresent()) {
             settings.readPreference(ReadPreference.valueOf(config.readPreference.get()));
         }
+
+        if (!connectionPoolListeners.isEmpty()) {
+            settings.applyToConnectionPoolSettings(builder -> {
```

Comment:
Please change this to not use lambdas. We try very hard to not use lambdas in runtime code because a small startup time and have a small memory footprint